base/show.jl

```diff
@@ -341,6 +341,8 @@ getindex(io::IOContext, key) = getindex(io.dict, key)
 getindex(io::IO, key) = throw(KeyError(key))
 get(io::IOContext, key, default) = get(io.dict, key, default)
 get(io::IO, key, default) = default
+keys(io::IOContext) = keys(io.dict)
+keys(io::IO) = ()
```

Comment:
Perhaps this should be `keys(ImmutableDict{Symbol,Any})`, but an empty tuple should be fine.

Reply:
Yes, I think it's worthwhile to make it return the same type. Let's change that and merge this.